core/src/main/kotlin/org/evomaster/core/problem/external/service/ExternalServices.kt

```diff
@@ -1,26 +1,169 @@
 package org.evomaster.core.problem.external.service
 
+import com.alibaba.dcm.DnsCacheManipulator
+import com.github.tomakehurst.wiremock.WireMockServer
+import com.github.tomakehurst.wiremock.core.WireMockConfiguration
+import com.github.tomakehurst.wiremock.extension.responsetemplating.ResponseTemplateTransformer
+import com.google.inject.Inject
+import org.evomaster.core.EMConfig
+import org.evomaster.core.problem.external.service.ExternalServiceUtils.generateRandomIPAddress
+import org.evomaster.core.problem.external.service.ExternalServiceUtils.isAddressAvailable
+import org.evomaster.core.problem.external.service.ExternalServiceUtils.nextIPAddress
+
+import com.github.tomakehurst.wiremock.client.WireMock.*
+
 class ExternalServices {
     /**
      * This will hold the information about the external service
      * calls inside the SUT. Information will be passed to the core
      * through AdditionalInfoDto and will be captured under
      * AbstractRestFitness and AbstractRestSample for further use.
+     *
+     * TODO: This is not the final implementation need to refactor but
+     * the concept is working.
      */
 
     /**
      * Contains the information about each external calls made
      */
     private val externalServiceInfos: MutableList<ExternalServiceInfo> = mutableListOf()
 
+    private var lastIPAddress : String = ""
+
+    @Inject
+    private lateinit var config : EMConfig
+
     /**
-     * This will allow adding ExternalServiceInfo to the set
+     * This will allow adding ExternalServiceInfo to the Collection
      */
     fun addExternalService(externalServiceInfo: ExternalServiceInfo) {
+        // TODO: The below code intentionally commented out
+        if (config.externalServiceIPSelectionStrategy != EMConfig.ExternalServiceIPSelectionStrategy.NONE) {
+            val ip = getIP()
+            lastIPAddress = ip
+            val wm : WireMockServer = initWireMockServer(ip)
+            updateDNSCache(externalServiceInfo.remoteHostname, ip)
+            externalServiceInfo.assignWireMockServer(wm)
+        }
         externalServiceInfos.add(externalServiceInfo)
     }
 
+    /**
+     * Will return the next available IP address from the last know IP address
+     * used for external service.
+     */
+    private fun getNextAvailableAddress() : String {
+        val nextAddress: String = nextIPAddress(lastIPAddress)
+
+        if (isAddressAvailable(nextAddress, WIREMOCK_PORT)) {
+            return nextAddress
+        } else {
+            throw Exception(nextAddress.plus(" is not available for use"))
+        }
+    }
+
+    /**
+     * Returns the default IP address for external service initialisation.
+     * If the respective address and port is not available it'll throw an
+     * exception.
+     */
+    private fun getDefaultAddress() : String {
+        if (isAddressAvailable("127.0.0.10", WIREMOCK_PORT)) {
+            return "127.0.0.10"
```

Comment:
the "default" in EMConfig is different, ie, 127.0.0.2. is there any reason for this?

Reply:
This was changed during a test, failed to revert back.